PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.webapp/src/com/microsoft/azuretools/webapp/ui/AppServiceCreateDialog.java

```diff
@@ -1022,17 +1022,19 @@ public static void initAspCache() {
         try {
             Map<String, List<AppServicePlan>> map = new ConcurrentHashMap<>();
             Set<SubscriptionDetail> sdl = AzureModel.getInstance().getSubscriptionToResourceGroupMap().keySet();
-            Observable.from(sdl).flatMap((sd) ->
-                Observable.create((subscriber) -> {
-                    try {
-                        List<AppServicePlan> appServicePlans = AzureWebAppMvpModel.getInstance()
-                            .listAppServicePlanBySubscriptionId(sd.getSubscriptionId());
-                        map.put(sd.getSubscriptionId(), appServicePlans);
-                        subscriber.onCompleted();
-                    } catch (Exception e) {
-                        Exceptions.propagate(e);
-                    }
-                }).subscribeOn(Schedulers.io()), sdl.size()).subscribeOn(Schedulers.io()).toBlocking().subscribe();
+            if (sdl.size() > 0) {
```

Comment:
Do we need to execute `subscriber.onCompleted();` when sdl.size()==0?

Reply:
Do not need, if the sdl.size == 0 , the code will go to subscribe....
![image](https://user-images.githubusercontent.com/7624690/57591997-df997180-7566-11e9-99a5-4bb2e1cf3c6d.png)